lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -129,8 +156,13 @@ def join_bulk_responses(bulk_responses)
     end
 
     def bulk_send(bulk_body)
-      # Discard the URL
-      url, response = @pool.post("_bulk", nil, bulk_body)
+      params = {}
+      if upload_compression then 
+        params[:headers] = {"Content-Encoding" => "gzip"}
+        bulk_body = Gzip::compress(bulk_body)
```

Comment:
Hi @dinnot, what do you think about making the compression here be more streaming oriented. To avoid having two copies of the bulk_body (`bulk_body` + `bulk_actions`), we can just keep a streaming `bulk_body` that is either a compressed, or not compressed StringIO object.